portal-frontend/src/app/features/application-details/application-details.component.ts

```diff
@@ -55,12 +61,14 @@ export class ApplicationDetailsComponent implements OnInit, OnDestroy {
     this.$applicationDocuments.pipe(takeUntil(this.$destroy)).subscribe((documents) => {
       this.otherFiles = documents
         .filter((file) => (file.type ? this.otherFileTypes.includes(file.type.code) : true))
+        .filter((file) => file.source === DOCUMENT_SOURCE.APPLICANT)
         .sort((a, b) => {
           return a.uploadedAt - b.uploadedAt;
         });
 
       this.authorizationLetters = documents
         .filter((file) => file.type?.code === DOCUMENT_TYPE.AUTHORIZATION_LETTER)
+        .filter((file) => file.source === DOCUMENT_SOURCE.APPLICANT)
```

Comment:
Just curious if there is any specific reason that you call filter one after another instead of && operator?

Reply:
Nope, just what I thought of